astropy/table/table.py

```diff
@@ -3023,10 +3025,10 @@ def _is_mapping(obj):
         else:
             raise TypeError('Vals must be an iterable or mapping or None')
 
+        # Insert val at index for each column
         columns = self.TableColumns()
-        try:
```

Comment:
Putting the `try` into the loop is logically the same but ensures correctness.

Reply:
Yes, clearer.